common/src/main/java/org/opensearch/ml/common/transport/model/MLUpdateModelInput.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.transport.model;
+
+import lombok.Data;
+import lombok.Builder;
+import lombok.Getter;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.ml.common.connector.Connector;
+import org.opensearch.ml.common.model.MLModelConfig;
+import org.opensearch.ml.common.model.TextEmbeddingModelConfig;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.ml.common.connector.Connector.createConnector;
+
+@Data
+public class MLUpdateModelInput implements ToXContentObject, Writeable {
+    
+    public static final String MODEL_ID_FIELD = "model_id"; // mandatory
+    public static final String DESCRIPTION_FIELD = "description"; // optional
+    public static final String MODEL_NAME_FIELD = "name"; // optional
+    public static final String MODEL_GROUP_ID_FIELD = "model_group_id"; // optional
+    public static final String MODEL_CONFIG_FIELD = "model_config"; // optional
+    public static final String CONNECTOR_ID_FIELD = "connector_id"; // optional
+
+    @Getter
+    private String modelId;
+    private String description;
+    private String name;
+    private String modelGroupId;
+    private MLModelConfig modelConfig;
+    private String connectorId;
+
+    @Builder(toBuilder = true)
+    public MLUpdateModelInput(String modelId, String description, String name, String modelGroupId, MLModelConfig modelConfig, String connectorId) {
+        this.modelId = modelId;
```

Comment:
Add some validation? For example the model id is mandatory, What if all fields are null except `modelId`?

Reply:
I think that's unnecessary. Update the model with an empty body is ok due to nothing has changed. It will do no harm to anything. (OS built-in update allows update with an empty body)